index.js

```diff
@@ -7,13 +7,26 @@ const YAML = require('js-yaml')
 const TOML = { parse: require('toml-j0.4').parse, stringify: require('tomlify-j0.4') }
 const INI = require('ini')
 
+/**
+ * Removes '.' in file extensions
+ * @param  {string} type input file extension
+ * @return {string}      file extension without '.'
+ */
+const validType = (type) => {
+    if (type.charAt(0) === '.') {
+        type = type.substr(1)
+    }
+    return type
+}
+
 const parse = (string, type) => {
     if (!string) throw new Error('missing string')
     if (!type) throw new Error('missing type')
+    type = validType(type)
     switch (type) {
         case 'json': return JSON.parse(string)
         case 'json5': return JSON5.parse(string)
-        case 'hjson': return HJSON.parse(string)
+        case 'hjson': return HJSON.parse(string.toString())
```

Comment:
naml/test.js
``` javascript
#!/usr/bin/env node
'use strict'

const HJSON = require('hjson')
const { readFiles } = require('./test/helpers')
const path = require('path')

var file = './test/cli/assets/input.hjson'
console.log('file: '+path.join(process.cwd(), file)+'\n');

readFiles([file])
.then(([string]) => {
  console.log(`HJSON.parse(string):`);
  console.log(
    HJSON.parse(string)
  )
})
.catch((err) => {
  console.error(err)
})

readFiles([file])
.then(([string]) => {
  console.log(`HJSON.parse(string.toString()):`);
  console.log(
    HJSON.parse(string.toString())
  )
})
.catch((err) => {
  console.error(err)
})


```

./test.js
``` console
file: /media/lil/data/Projects/naml-2/test/cli/assets/input.hjson


HJSON.parse(string):
Error: source is not a string
    at Object.module.exports [as parse] (/media/lil/data/Projects/naml-2/node_modules/hjson/lib/hjson-parse.js:443:39)
    at readFiles.then (/media/lil/data/Projects/naml-2/test.js:15:11)

HJSON.parse(string.toString()):
{ name: 'naml',
  version: '1.0.5',
  description: 'Not Another Markup Language',
  bin: { naml: 'cli.js' },
  main: 'index.js',
  scripts:
   { pretest: 'npm link',
     test: 'nyc ava test/*/*.js --serial --verbose' },
  engines: { node: '>=6.0.0' },
  keywords:
   [ 'naml',
     'json',
     'json5',
     'hjson',
     'cson',
     'yaml',
     'toml',
     'ini',
     'parse',
     'stringify',
     'convert' ],
  author: 'Mark Tiedemann <www.marktiedemann@gmail.com>',
  license: 'WTFPL',
  dependencies:
   { chalk: '^1.1.3',
     cson: '^4.1.0',
     'fs-extra': '^4.0.0',
     hjson: '^3.1.0',
     ini: '^1.3.4',
     'js-yaml': '^3.9.0',
     json5: '^0.5.1',
     'toml-j0.4': '^1.1.0',
     'tomlify-j0.4': '^2.2.0' },
  devDependencies: { ava: '^0.15.2', nyc: '^6.4.4' },
  directories: { test: 'test' },
  repository:
   { type: 'git',
     url: 'git+https://github.com/MarkTiedemann/naml.git' },
  bugs: { url: 'https://github.com/MarkTiedemann/naml/issues' },
  homepage: 'https://github.com/MarkTiedemann/naml' }

```

for more info see:
- `test/helpers.js`
- `test/cli/convert.js`